File: JavaScript/264-Ugly Number II.js
/**
 * @param {number} n
 * @return {number}
 */
var nthUglyNumber = function(n) {
    let i2=0, i3=0, i5=0;
    let arr = [1];
    
    while (arr.length < n){
        arr.push(Math.min(2*arr[i2],3*arr[i3],5*arr[i5]));
        if (arr[arr.length-1] === 2*arr[i2]) i2+=1;
        if (arr[arr.length-1] === 3*arr[i3]) i3+=1;
        if (arr[arr.length-1] === 5*arr[i5]) i5+=1;
    }
    
    return arr[arr.length-1];
};

File: C++/264-Ugly Number II.cpp
class Solution {
public:
    int nthUglyNumber(int n) {
        vector<int> res = {1};
        int i2 = 0, i3 = 0, i5 = 0;
        for (int i=1; i<n; i++){
            // pick the added ugly numbers that have not multiply by 2 or 3 or 5 yet
            int nxt = min(res[i2]*2, min(res[i3]*3, res[i5]*5));   
            if (nxt == res[i2]*2) i2++;
            if (nxt == res[i3]*3) i3++;
            if (nxt == res[i5]*5) i5++;
            res.push_back(nxt);
        }
        return res[res.size()-1];
    }
};